fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanReadPlatformServiceImpl.java

```diff
@@ -431,7 +431,7 @@ public LoanTransactionData retrieveLoanTransactionTemplate(final Long loanId) {
         this.context.authenticatedUser();
 
         RepaymentTransactionTemplateMapper mapper = new RepaymentTransactionTemplateMapper();
-        String sql = "select " + mapper.schema() + " where l.id =?";
+        String sql = "select " + mapper.schema();
         LoanTransactionData loanTransactionData = this.jdbcTemplate.queryForObject(sql, mapper, LoanTransactionType.REPAYMENT.getValue(),
```

Comment:
If we are removing one parameter from the query, should we not also pass one less parameter when we execute the query?

Reply:
I did not remove the parameter i only put it in its rightful place 
            sqlBuilder.append(" GROUP BY ls.duedate");	            sqlBuilder.append("WHERE l.id = ? ");
just before the groub by object